### ISSUE TITLE
Feature request - Fetch code_hash without the entire ABI

### ISSUE THREAD

**Issue by nsjames** [EOSIO/eos#4222]
Some ABIs are massive ( system for instance is 1.3mb ) and take a good 1-3 seconds to fetch. 
If there was a way to fetch only the hash of the contract without the rest of the ABI it would allow signature providers a way to cache ABIs locally and only re-cache if the code hash changes. 

`eosjs` doesn't propagate the code abi to the signature provider so it has to be double fetched ( once for eosjs and once for the signature app ) to parse data and retrieve ricardian contracts which adds between 2 - 6 seconds per contract from the moment a user clicks a button to sign to the time the transaction parameters + ricardians can be displayed to them.

**Comment by nsjames** [comment]
Actually seeing as how getting only the hash is supported in cleos, I'm moving this ticket to `eosjs` itself for support there.

**Comment by nsjames** [comment]
Reopening as it seems the RPC doesn't have an option for code_hash only 
https://github.com/EOSIO/eos/blob/master/plugins/chain_plugin/chain_plugin.cpp#L952

**Comment by jcalfee** [comment]
Did you see this:  https://github.com/EOSIO/eos/issues/566

Looks like get_abi was added about 2 weeks ago.. so that will help greatly.  I was foced to fetch code and the ABI at the same time.  I'll update to use get_abi on this ticket.  

It will not tell me if it was modified though so I will have to keep re-fetching..

**Comment by nsjames** [comment]
Yeah I saw that one last night too, which is massively faster than get_code, but doesn't include the hash :(

**Comment by jcalfee** [comment]
I suggest adding abi_hash:

```c++
   fc::sha256 abi_hash;
```
Then add `optional<abi_hash> abi_hash` to get_abi's parameters and result:

```c++
   struct get_abi_params {
      name account_name;
      optional<abi_hash> abi_hash;
   };

   struct get_abi_results {
      name                   account_name;
      optional<abi_def>      abi;
      optional<abi_hash>     abi_hash;
   };
```
* The `abi_hash` is the sha256 hash of the binary `abi_def` serialization
* If an ABI exists for the account its hash is always returned.
* If a hash was provided in the request and it matches the current ABI, the ABI hash is returned but the abi is not.
* If a hash was provided in the request and the ABI does not exist (was probably removed) neither hash nor ABI are returned.

**Comment by nsjames** [comment]
@tbfleming

**Comment by ScottSallinen** [comment]
For getting just the code hash, see this PR:
https://github.com/EOSIO/eos/pull/5434

**Comment by jcalfee** [comment]
Great  .. I'll have it in eoejs for the next release.

**Comment by tbfleming** [comment]
@jcalfee I added your abi_hash caching idea to #5375's `get_raw_abi`

**Comment by jgiszczak** [comment]
The `get_code_hash` API has been available since the v1.3.0 release on 2018 September 18.
